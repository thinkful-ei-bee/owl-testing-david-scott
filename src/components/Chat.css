.chatEvent {
  color: grey;
  text-align: center;
}

.chatEvent img {
  border-radius: 50%;
  display: inline-block;
  width: 30px;
  height: 30px;
}

.sideBar {
  float: left;
  min-height: 600px;
  width: 300px;
  overflow: scroll;
}

.timeStamped {
  text-align: left;
}

.indented {
  color: black;
  text-indent: 25px;
}

.sideBarContainer {
  width: 300px;
}

input {
  margin-left: 10px;
  margin-top: 10px;
  height: 40px;
  width: 90%;
}