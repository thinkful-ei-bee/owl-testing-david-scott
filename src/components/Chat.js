import React from 'react';
import './Chat.css';

class Chat extends React.Component {
  render() {
    const eventMapping = {
      'thumbs-up': 'gave a thumbs up',
      'thumbs-down': 'gave a thumbs down',
      'clap': 'clapped',
      'raise-hand': 'raised their hand',
      'join': 'joined', 
      'leave': 'left',
      'join-stage': 'joined the stage',
      'leave-stage': 'left the stage'
    }
    const participants = this.props.participants;
    const chatEvents = this.props.chatEvents;
    const chatTextBoxes = chatEvents.map((evt, i) => {
      const speaker = participants.find(ele => ele.id === evt.participantId);
      if(evt.type === 'message') {
          const messageTime = new Date(evt.timestamp);
          const convertedTime = messageTime.toLocaleString('en-US', {
            hour: 'numeric',
            minute: 'numeric',
            hour12: true
          })
          return (
            <div className="chatEvent timeStamped" key={i}>
              <img src={speaker.avatar} alt={speaker.name} />
              <strong>{speaker.name} </strong>
              <span> {convertedTime}</span>
              <p class="indented">{evt.message}</p>
            </div>
          );
      } else {
        return (
          <div className="chatEvent" key={i}>
              <p><strong>{speaker.name}</strong> {eventMapping[evt.type]}</p>
          </div>
        );
      } 
       
    });
  
    return (
      <div className="sideBarContainer">
         <div className="sideBar">
            <div className="allMessages">{chatTextBoxes}</div>
        </div>
        <input type="text" placeholder="Chat"></input>
      </div>
    );
  }
}

export default Chat;

