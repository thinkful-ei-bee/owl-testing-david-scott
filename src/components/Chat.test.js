import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Chat from './Chat';

const participants = [
  {
      id: 1,
      name: 'Koren Templeton',
      avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  },
  {
      id: 2,
      name: 'Caty Flucker',
      avatar:
          'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: false
  },
  {
      id: 3,
      name: 'Axe Kubicka',
      avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
      inSession: false,
      onStage: true
  }];

const chatEvents = [
  {
   participantId: 3,
   type: 'message',
   message: 'Hello world',
   time: 1548852646559,
   timestamp: 1548852484247
}];

describe('<Chat />', () => {
  it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Chat chatEvents={chatEvents} participants={participants}/>, div);
  ReactDOM.unmountComponentAtNode(div);
  });

  it('matches previous snapshot', () => {
    const tree = renderer.create(<Chat chatEvents={chatEvents} participants={participants} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});

