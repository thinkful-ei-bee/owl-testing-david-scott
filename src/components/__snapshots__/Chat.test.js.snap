// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> matches previous snapshot 1`] = `
<div
  className="sideBarContainer"
>
  <div
    className="sideBar"
  >
    <div
      className="allMessages"
    >
      <div
        className="chatEvent timeStamped"
      >
        <img
          alt="Axe Kubicka"
          src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
        />
        <strong>
          Axe Kubicka
           
        </strong>
        <span>
           
          4:48 AM
        </span>
        <p
          class="indented"
        >
          Hello world
        </p>
      </div>
    </div>
  </div>
  <input
    placeholder="Chat"
    type="text"
  />
</div>
`;
